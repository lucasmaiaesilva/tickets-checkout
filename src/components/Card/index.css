.card {
  background-color: #f3f3f3;
  border: 1px solid #d1d5da;
  border-radius: 3px;
  box-shadow: 0 2px 12px 0 rgba(0,0,0,.1);
}

.card .detail {
  padding: 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.card .detail p {
  margin: 12px 0;
}

.card .detail .price>span:first-child {
  margin-right: 5px;
  font-size: 1.4rem;
}

.card .detail .price span {
  font-weight: bold;
}

.card .description {
  color: rgb(138, 163, 187);
  font-weight: bold;
  padding: 10px;
  border-top: 1px solid #d1d5da;
}

