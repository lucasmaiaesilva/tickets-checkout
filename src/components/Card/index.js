import React from 'react';
import TicketDescription from '../TicketDescription';
import Select from '../Select';

import './index.css';

const Card = ({
  number,
  price,
  taxes,
  quantities,
  handleChange,
  id,
  isActive,
  discount,
  description
}) => (
  <div className="card">
    <div className="detail">
      <div className="block">
        <p>Royal - lote {number}</p>
        <p className="price">
          <span>R$ {(price / 100).toFixed(2).replace('.', ',')}</span>
          <span>(+ R$ {(taxes / 100).toFixed(2)} em taxas)</span>
        </p>
      </div>
      <Select
        quantities={quantities}
        handleChange={handleChange}
        ticketId={id}
        price={price}
        disabled={!isActive}
        discount={discount}
      />
    </div>
    <TicketDescription text={description}  />
  </div> 
);

export default Card;

