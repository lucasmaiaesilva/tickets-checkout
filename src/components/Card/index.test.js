import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './';

describe('[Component] Card', () => {
  const props = {
    number: 1,
    price: 1000,
    taxes: 1,
    quantities: [1, 2, 3],
    handleChange: jest.fn(),
    id: 'asd',
    discount: 1,
    description: 'hello world',
  };
  it('render inactive', () => {
    const wrapper = shallow(<Card {...props} disabled />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render Active card', () => {
    const wrapper = shallow(<Card {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

