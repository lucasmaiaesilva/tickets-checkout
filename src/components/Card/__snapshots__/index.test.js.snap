// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Card render Active card 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    description="hello world"
    discount={1}
    handleChange={[Function]}
    id="asd"
    number={1}
    price={1000}
    quantities={
        Array [
            1,
            2,
            3,
          ]
    }
    taxes={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="detail"
>
          <div
                    className="block"
          >
                    <p>
                              Royal - lote 
                              1
                    </p>
                    <p
                              className="price"
                    >
                              <span>
                                        R$ 
                                        10,00
                              </span>
                              <span>
                                        (+ R$ 
                                        0.01
                                         em taxas)
                              </span>
                    </p>
          </div>
          <Select
                    disabled={true}
                    discount={1}
                    handleChange={[Function]}
                    price={1000}
                    quantities={
                              Array [
                                        1,
                                        2,
                                        3,
                                      ]
                    }
                    ticketId="asd"
          />
</div>,
        <TicketDescription
          text="hello world"
/>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="block"
>
              <p>
                            Royal - lote 
                            1
              </p>
              <p
                            className="price"
              >
                            <span>
                                          R$ 
                                          10,00
                            </span>
                            <span>
                                          (+ R$ 
                                          0.01
                                           em taxas)
                            </span>
              </p>
</div>,
            <Select
              disabled={true}
              discount={1}
              handleChange={[Function]}
              price={1000}
              quantities={
                            Array [
                                          1,
                                          2,
                                          3,
                                        ]
              }
              ticketId="asd"
/>,
          ],
          "className": "detail",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Royal - lote 
                  1
</p>,
                <p
                  className="price"
>
                  <span>
                                    R$ 
                                    10,00
                  </span>
                  <span>
                                    (+ R$ 
                                    0.01
                                     em taxas)
                  </span>
</p>,
              ],
              "className": "block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Royal - lote ",
                    1,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Royal - lote ",
                  1,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      R$ 
                      10,00
</span>,
                    <span>
                      (+ R$ 
                      0.01
                       em taxas)
</span>,
                  ],
                  "className": "price",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "R$ ",
                        "10,00",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "R$ ",
                      "10,00",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "(+ R$ ",
                        "0.01",
                        " em taxas)",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "(+ R$ ",
                      "0.01",
                      " em taxas)",
                    ],
                    "type": "span",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": true,
              "discount": 1,
              "handleChange": [Function],
              "price": 1000,
              "quantities": Array [
                1,
                2,
                3,
              ],
              "ticketId": "asd",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "text": "hello world",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="detail"
>
            <div
                        className="block"
            >
                        <p>
                                    Royal - lote 
                                    1
                        </p>
                        <p
                                    className="price"
                        >
                                    <span>
                                                R$ 
                                                10,00
                                    </span>
                                    <span>
                                                (+ R$ 
                                                0.01
                                                 em taxas)
                                    </span>
                        </p>
            </div>
            <Select
                        disabled={true}
                        discount={1}
                        handleChange={[Function]}
                        price={1000}
                        quantities={
                                    Array [
                                                1,
                                                2,
                                                3,
                                              ]
                        }
                        ticketId="asd"
            />
</div>,
          <TicketDescription
            text="hello world"
/>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="block"
>
                <p>
                                Royal - lote 
                                1
                </p>
                <p
                                className="price"
                >
                                <span>
                                                R$ 
                                                10,00
                                </span>
                                <span>
                                                (+ R$ 
                                                0.01
                                                 em taxas)
                                </span>
                </p>
</div>,
              <Select
                disabled={true}
                discount={1}
                handleChange={[Function]}
                price={1000}
                quantities={
                                Array [
                                                1,
                                                2,
                                                3,
                                              ]
                }
                ticketId="asd"
/>,
            ],
            "className": "detail",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Royal - lote 
                    1
</p>,
                  <p
                    className="price"
>
                    <span>
                                        R$ 
                                        10,00
                    </span>
                    <span>
                                        (+ R$ 
                                        0.01
                                         em taxas)
                    </span>
</p>,
                ],
                "className": "block",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Royal - lote ",
                      1,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Royal - lote ",
                    1,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        R$ 
                        10,00
</span>,
                      <span>
                        (+ R$ 
                        0.01
                         em taxas)
</span>,
                    ],
                    "className": "price",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "R$ ",
                          "10,00",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "R$ ",
                        "10,00",
                      ],
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "(+ R$ ",
                          "0.01",
                          " em taxas)",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "(+ R$ ",
                        "0.01",
                        " em taxas)",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": true,
                "discount": 1,
                "handleChange": [Function],
                "price": 1000,
                "quantities": Array [
                  1,
                  2,
                  3,
                ],
                "ticketId": "asd",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "text": "hello world",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`[Component] Card render inactive 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    description="hello world"
    disabled={true}
    discount={1}
    handleChange={[Function]}
    id="asd"
    number={1}
    price={1000}
    quantities={
        Array [
            1,
            2,
            3,
          ]
    }
    taxes={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="detail"
>
          <div
                    className="block"
          >
                    <p>
                              Royal - lote 
                              1
                    </p>
                    <p
                              className="price"
                    >
                              <span>
                                        R$ 
                                        10,00
                              </span>
                              <span>
                                        (+ R$ 
                                        0.01
                                         em taxas)
                              </span>
                    </p>
          </div>
          <Select
                    disabled={true}
                    discount={1}
                    handleChange={[Function]}
                    price={1000}
                    quantities={
                              Array [
                                        1,
                                        2,
                                        3,
                                      ]
                    }
                    ticketId="asd"
          />
</div>,
        <TicketDescription
          text="hello world"
/>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="block"
>
              <p>
                            Royal - lote 
                            1
              </p>
              <p
                            className="price"
              >
                            <span>
                                          R$ 
                                          10,00
                            </span>
                            <span>
                                          (+ R$ 
                                          0.01
                                           em taxas)
                            </span>
              </p>
</div>,
            <Select
              disabled={true}
              discount={1}
              handleChange={[Function]}
              price={1000}
              quantities={
                            Array [
                                          1,
                                          2,
                                          3,
                                        ]
              }
              ticketId="asd"
/>,
          ],
          "className": "detail",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Royal - lote 
                  1
</p>,
                <p
                  className="price"
>
                  <span>
                                    R$ 
                                    10,00
                  </span>
                  <span>
                                    (+ R$ 
                                    0.01
                                     em taxas)
                  </span>
</p>,
              ],
              "className": "block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Royal - lote ",
                    1,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Royal - lote ",
                  1,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      R$ 
                      10,00
</span>,
                    <span>
                      (+ R$ 
                      0.01
                       em taxas)
</span>,
                  ],
                  "className": "price",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "R$ ",
                        "10,00",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "R$ ",
                      "10,00",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "(+ R$ ",
                        "0.01",
                        " em taxas)",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "(+ R$ ",
                      "0.01",
                      " em taxas)",
                    ],
                    "type": "span",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": true,
              "discount": 1,
              "handleChange": [Function],
              "price": 1000,
              "quantities": Array [
                1,
                2,
                3,
              ],
              "ticketId": "asd",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "text": "hello world",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="detail"
>
            <div
                        className="block"
            >
                        <p>
                                    Royal - lote 
                                    1
                        </p>
                        <p
                                    className="price"
                        >
                                    <span>
                                                R$ 
                                                10,00
                                    </span>
                                    <span>
                                                (+ R$ 
                                                0.01
                                                 em taxas)
                                    </span>
                        </p>
            </div>
            <Select
                        disabled={true}
                        discount={1}
                        handleChange={[Function]}
                        price={1000}
                        quantities={
                                    Array [
                                                1,
                                                2,
                                                3,
                                              ]
                        }
                        ticketId="asd"
            />
</div>,
          <TicketDescription
            text="hello world"
/>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="block"
>
                <p>
                                Royal - lote 
                                1
                </p>
                <p
                                className="price"
                >
                                <span>
                                                R$ 
                                                10,00
                                </span>
                                <span>
                                                (+ R$ 
                                                0.01
                                                 em taxas)
                                </span>
                </p>
</div>,
              <Select
                disabled={true}
                discount={1}
                handleChange={[Function]}
                price={1000}
                quantities={
                                Array [
                                                1,
                                                2,
                                                3,
                                              ]
                }
                ticketId="asd"
/>,
            ],
            "className": "detail",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Royal - lote 
                    1
</p>,
                  <p
                    className="price"
>
                    <span>
                                        R$ 
                                        10,00
                    </span>
                    <span>
                                        (+ R$ 
                                        0.01
                                         em taxas)
                    </span>
</p>,
                ],
                "className": "block",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Royal - lote ",
                      1,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Royal - lote ",
                    1,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        R$ 
                        10,00
</span>,
                      <span>
                        (+ R$ 
                        0.01
                         em taxas)
</span>,
                    ],
                    "className": "price",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "R$ ",
                          "10,00",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "R$ ",
                        "10,00",
                      ],
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "(+ R$ ",
                          "0.01",
                          " em taxas)",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "(+ R$ ",
                        "0.01",
                        " em taxas)",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": true,
                "discount": 1,
                "handleChange": [Function],
                "price": 1000,
                "quantities": Array [
                  1,
                  2,
                  3,
                ],
                "ticketId": "asd",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "text": "hello world",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
